packages/server/src/github/api.js

```diff
@@ -0,0 +1,231 @@
+const assert = require('assert');
+const _ = require('lodash');
+const request = require('request');
+const crypto = require('crypto');
+
+const config = require('../config');
+
+class API {
+  /**
+   * @public
+   */
+  constructor() {
+    this.repoOwner = config.GITHUB_REPO_OWNER;
+    this.repoName = config.GITHUB_REPO_NAME;
+    this.token = config.GITHUB_PERSONAL_ACCESS_TOKEN;
+    this.webhookSecret = config.GITHUB_WEBHOOK_SECRET;
+    this.apiURL = `https://api.github.com`;
+    this.serverURL = config.SERVER_URL;
+    this.hookUrl = `${this.serverURL}/hooks`;
+    this.isReady = true;
+
+    if (!this.repoOwner) {
+      console.warn(`GitHub repo owner is not set.`);
+      this.isReady = false;
+    }
+    if (!this.repoName) {
+      console.warn(`GitHub repo name is not set.`);
+      this.isReady = false;
+    }
+    if (!this.token) {
+      console.warn(
+        `GitHub personal access token is not set, ` +
+        `see https://github.com/settings/tokens.`
+      );
+      this.isReady = false;
+    }
+    if (!this.webhookSecret) {
+      console.warn(`GitHub webhook secret is not set`);
+      this.isReady = false;
+    }
```

Comment:
Again, I feel like the top 20 lines could be dried up with a pattern. Like,
``validate(!this.token, "Github personal access token is not set")``

Reply:
On that note, the text messages being here is a bit sloppy. We could have a file or at least a {} lib for text.

``ERROR_TEXT = {
NO_OWNER = `GitHub repo owner is not set.`
...
}

Reply:
This messages are aimed only for developers, it is not a public part of the application. So this is why here is no need to put it to somewhere to place it in one place. These part of code is stuff. Today this part is tomorrow is not. So I believe that the better approach here is to "inline" code instead of using magic and help functions.

Reply:
Have added a little refactoring here.

Reply:
So even If I use a dictionary with these messages for developers I will have to do it over the whole application to make it consistency and do not confuse developers.

Reply:
I'd agree you'd want consistency, which is why I recommended eventually just making a new story/chore for doing this in a broader way.

Reply:
Anyway it is not ordinary way to store dev messages in one place, need to review another applications to make decision.

Reply:
And another argument that it will raise cohesion between encapsulated modules (https://en.wikipedia.org/wiki/Cohesion_(computer_science)), so let's stay it this way may be for a short time.